File: LeetCode/python3/28.py
class Solution:
    def strStr(self, haystack: str, needle: str) -> int:
        n, m = len(haystack), len(needle)
        if m == 0:
            return 0
        for i in range(n):
            if i + m > n:
                return -1
            j = 0
            while i + j < n and j < m and haystack[i + j] == needle[j]:
                j += 1
            if j == m:
                return i
        return -1

File: LeetCode/cpp/28.cpp
class Solution {
public:
    int strStr(string &haystack, string &needle) {
        int n = haystack.size(), m = needle.size();
        if (m == 0)
            return 0;
        for (int i = 0; i < n; ++i) {
            if (i + m > n)
                return -1;
            int j = 0;
            while (j < m && haystack[i + j] == needle[j])
                ++j;
            if (j == m)
                return i;
        }
        return -1;
    }
};
